src/lib/platform/posix/os-types.h

```diff
@@ -44,6 +44,8 @@
 #include <poll.h>
 #include <semaphore.h>
 #include <stdint.h>
+#include <dirent.h>
```

Comment:
libplatform is used by lots of things, and these headers are not used by the platform lib itself. also, this got moved into a separate library now